In [1]:
import pandas as pd

In [2]:
airbnb = pd.read_csv('airbnb_data.csv')

In [5]:
%%sql
SELECT 
    room_type,
    COUNT(*) AS num_listings
FROM 
    airbnb
GROUP BY 
    room_type
ORDER BY 
    num_listings DESC
LIMIT 1;

room_type  num_listings
0  Entire place          4075

In [6]:
%%sql
SELECT 
    room_type,
    AVG(rating) AS avg_rating,
    AVG(price) AS avg_price
FROM 
    airbnb
GROUP BY 
    room_type;

room_type  avg_rating   avg_price
0  Private Room    4.010318   81.928449
1  Entire place    4.015791  193.170798
2   Shared room    3.963321   68.006061

In [9]:
%%sql
SELECT 
    neighbourhood_full,
    room_type,
    COUNT(*) AS num_listings
FROM 
    airbnb
GROUP BY 
    neighbourhood_full, room_type
ORDER BY 
    num_listings DESC
LIMIT 10;

neighbourhood_full     room_type  num_listings
0  Brooklyn, Bedford-Stuyvesant  Private Room           336
1        Brooklyn, Williamsburg  Entire place           309
2        Brooklyn, Williamsburg  Private Room           293
3  Brooklyn, Bedford-Stuyvesant  Entire place           286
4            Brooklyn, Bushwick  Private Room           275
5             Manhattan, Harlem  Private Room           251
6    Manhattan, Upper East Side  Entire place           208
7             Manhattan, Harlem  Entire place           186
8    Manhattan, Upper West Side  Entire place           182
9       Manhattan, East Village  Entire place           176

Most Common Room Type:

- Identified the most common room type among Airbnb listings in the dataset.
- Counted the occurrences of each room type and selected the most frequent one.
- Result: The most common room type is "Entire place", with 4,075 listings.

Average Ratings and Prices by Room Type:

- Calculated the average rating and average price for each room type.
- Grouped the listings by room type and compute the average rating and price.
- Results:
  * Private Room: Average Rating = 4.01, Average Price = $81.93
  * Entire Place: Average Rating = 4.02, Average Price = $193.17
  * Shared Room: Average Rating = 3.96, Average Price = $68.01

Popularity of Room Types in Specific Neighborhoods:

- Examined the popularity of different room types within specific neighborhoods.
- Counted the occurrences of each room type within each neighborhood and ranked them by the number of listings.
- Identified the top 10 combinations of neighborhood and room type with the highest number of listings.

These analyses provide insights into the distribution, ratings, prices, and popularity of different room types among Airbnb listings, both overall and within specific neighborhoods.